### PR TITLE
(SLV-152) Update CLI to match spec

### PR DESCRIPTION
**Pull request by RandellP**
reconfigured to have an install command and several subcommands.
Added stubs for the new subcommands
Setup install to call all of the subcommands as approriate.

### CODE REVIEW

File: bin/ref_arch_setup [hunk 1]
```diff
@@ -33,37 +37,124 @@ BANNER
...
+    @options["pe_version"] = value
+  end
+  opt.on("--console-password value", "password for the PE console") do |value|
+    @options["console_password"] = value
+  end
+end
+
+def install_bootstrap(opt)
+  # TODO: remove primary master once we get implement getting the host from the pe.conf
+  opt.on("--primary-master value", "hostname of primary master") do |value|
```
Comment (billclaytor) [inline bin/ref_arch_setup]: Capitalization for the descriptions should be consistent.

File: bin/ref_arch_setup [hunk 2]
```diff
@@ -33,37 +37,124 @@ BANNER
...
+  opt.on("--password value", "passworf for bolt to use to connect to target hosts") do |value|
+    @options["password"] = value
+  end
+  opt.on("--private-key value", "private key for bolt to use to connect to target hosts") do |value|
+    @options["private_key"] = value
+  end
+end
+
+def install_generate_opts(opt)
+  opt.on("--primary-master value", "hostname of primary master") do |value|
```
Comment (billclaytor) [inline bin/ref_arch_setup]: Should we indicate required parameters, perhaps by either replacing "value" with "required" or "optional", or starting the description with "(required)"?

**Comment by RandellP** [comment]
@billclaytor I pushed an update to address the feedback.  Also the your feedback and JD's (UX) got me thinking on some things that caused me to make more changes to the overall output text.  I ran all the help commands and pasted the output in the interface doc. https://docs.google.com/document/d/1vz8JcNrXSWAucKxZ0yPEOBNKFO2Wi9DUYI-KQSLK-sY/edit?ts=5b299c7b#  For easier reading. :)

File: bin/ref_arch_setup [hunk 3]
```diff
@@ -33,37 +37,170 @@ BANNER
...
-    opt.banner = "Usage: ref_arch_setup install [options]"
-    opt.on("--target-host value", "Host to install PE on") do |value|
-      options["target_host"] = value
-    end
-    opt.on("--pe-tarball-path value", "Path to PE tarball") do |value|
-      options["pe_tarball_path"] = value
-    end
-    opt.on("--pe-conf-path value", "Path to pe.conf file") do |value|
-      options["pe_conf_path"] = value
+    opt.banner = <<-BANNER
```
Comment (billclaytor) [inline bin/ref_arch_setup]: The option summary "--console-password <password>" forces the description onto the next line. Setting the summary width to at least 33 resolves the issue (we may want to add extra padding):
`opt.set_summary_width(33)`

File: bin/ref_arch_setup [hunk 4]
```diff
@@ -33,37 +37,170 @@ BANNER
...
+    Available Options:
+      Either --console-password or --pe-conf required
+    BANNER
+    install_opts(opt)
+  end
+}
+
+subcommands = {
+  "generate-pe-conf" => OptionParser.new do |opt|
+    opt.banner = <<-BANNER
```
Comment (billclaytor) [inline bin/ref_arch_setup]: See above

**Pull request merged**